### DISCUSSION TITLE
Cannot run cypress with a fresh yarn pnp install

### DISCUSSION THREAD

**Discussion by saudpunjwani101** [cypress-io/cypress#27263] [Questions and Help]
I'm trying to setup Cypress with jest adapter and testing-library/cypress but I'm encountering an issue as I'm using Yarn Pnp. Here I tried to do it in a fresh project.

`mkdir test-cypress && cd test-cypress`
`yarn init`
`yarn add react react-dom vite`
`yarn add cypress --dev`
`yarn run cypress open`

And I get an error,

> Error: Your application tried to access @opentelemetry/resources, but it isn't declared in your dependencies; this make
> s the require call ambiguous and unsound.
> 
> Required package: @opentelemetry/resources (via "@opentelemetry/resources")
> Required by: /Users/Saud/Library/Caches/Cypress/12.17.1/Cypress.app/Contents/Resources/app/packages/telemetry/dist/

My package.json file,

`{
  "name": "new-cypress",
  "packageManager": "yarn@3.3.0",
  "dependencies": {
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "vite": "^4.4.2"
  },
  "devDependencies": {
    "cypress": "12.17.1"
  }
}
`

If I do it via npm, it works fine. The problem is somehow related to Yarn pnp. Note that I've tried downgrading cypress to v10 and v9. The same problem occurs, just that in v10 I get this same error but for a different library like "lodash" and in v9 I get this error for library "graceful"

**Comment by public** [comment]
Try [loose mode](https://yarnpkg.com/features/pnp#pnp-loose-mode). e.g. By setting `YARN_PNP_MODE=loose`

Alternatively try setting `nodeLinker: pnpm` in your `.yarnrc.yml` and removing all the `pnp` files. This will make your setup not fully Zero Install as it will have to unpack all the zip files from the cache or download them again but it does produce a sufficiently normal node_modules to keep Cypress happy.

**Comment by tyler-austin** [comment]
did this fix your issue? I am getting the same error.

**Comment by pondchamp** [comment]
Getting the same error here on Win11... I've been needing to set `nodeLinker: node-modules` in my .yarnrc.yml file temporarily as a workaround to this issue

**Reply by MikeMcC399** [reply]
- See https://github.com/cypress-io/cypress/issues/27542#issuecomment-1991216610 for resolution suggestions
